packages/@ourworldindata/core-table/src/CoreTableUtils.ts

```diff
@@ -548,16 +550,46 @@ export const matrixToDelimited = (
         .join("\n")
 }
 
+/**
+ * An array object representing all parsed rows. The array is enhanced with a property listing
+ * the names of the parsed columns.
+ */
+export interface DSVParsedArray<T> extends Array<T> {
+    /**
+     * List of column names.
+     */
+    columns: Array<keyof T>
+}
+
 export const parseDelimited = (
     str: string,
     delimiter?: string,
-    parseFn?: (
-        rawRow: Record<string, string | undefined> | undefined,
-        index: number,
-        columns: unknown[]
-    ) => Record<string, unknown> | undefined | null
-): DSVParsedArray<Record<string, unknown>> =>
-    dsvFormat(delimiter ?? detectDelimiter(str)).parse(str, parseFn as any)
+    parseFn?: (rawRow: RawRow) => ParsedRow
+): DSVParsedArray<Record<string, any>> => {
+    // Convert PapaParse result to D3 format for backwards compatibility
+    const papaParseToD3 = ({
+        data,
+        meta,
+    }: Papa.ParseResult<any>): DSVParsedArray<Record<string, any>> => {
+        const dsvParsed = data as DSVParsedArray<Record<string, any>>
+        dsvParsed.columns = meta.fields || []
+        return dsvParsed
+    }
+
+    const result = Papa.parse(str, {
+        delimiter: delimiter ?? detectDelimiter(str),
+        skipEmptyLines: true,
```

Comment:
Probably better to leave this one out, since it's not the behaviour that `d3-dsv` showed afaict.

Reply:
We have a test for dropping empty lines which fails when I remove this line, so I think it is 🙂